embedding/csharp/Tizen.Flutter.Embedding/ElmFlutterView.cs

```diff
@@ -0,0 +1,189 @@
+        private int InitialWidth { get; set; } = 0;
+
+        /// <summary>
+        /// The initial height of the view. Defaults to the parent height if the value is zero.
+        /// </summary>
+        private int InitialHeight { get; set; } = 0;
+
+        /// <summary>
+        /// Whether the view is running.
+        /// </summary>
+        public bool IsRunning => !View.IsInvalid;
+
+        /// <summary>
+        /// The current width of the view.
+        /// </summary>
+        public int Width
+        {
+            get
+            {
+                Debug.Assert(IsRunning);
+
+                return EvasObject.Geometry.Width;
+            }
+        }
+
+        /// <summary>
+        /// The current height of the view.
+        /// </summary>
+        public int Height
+        {
+            get
+            {
+                Debug.Assert(IsRunning);
+
+                return EvasObject.Geometry.Height;
+            }
+        }
+
+        public ElmFlutterView(EvasObject parent)
+        {
+            Parent = parent;
+        }
+
+        public ElmFlutterView(EvasObject parent, int initialWidth, int initialHeight)
+        {
+            Parent = parent;
+            InitialWidth = initialWidth;
+            InitialHeight = initialHeight;
+        }
+
+        /// <summary>
+        /// Starts running the view with the associated engine, creating if not set.
+        /// </summary>
+        public bool RunEngine()
+        {
+            if (Parent == null)
+            {
+                TizenLog.Error("The parent object is invalid.");
+                return false;
+            }
+
+            if (Engine == null)
+            {
+                Engine = new FlutterEngine();
+            }
+
+            if (!Engine.IsValid)
+            {
+                throw new Exception("Could not create a Flutter engine.");
+            }
+
+            var viewProperties = new FlutterDesktopViewProperties
+            {
+                width = InitialWidth,
+                height = InitialHeight,
+            };
+
+            View = FlutterDesktopViewCreateFromElmParent(ref viewProperties, Engine.Engine, Parent);
+            if (View.IsInvalid)
+            {
+                TizenLog.Error("Could not launch a Flutter view.");
+                return false;
+            }
+
+            EvasObject = new EvasObjectImpl(Parent, FlutterDesktopViewGetEvasObject(View));
+            if (!EvasObject.IsRealized)
+            {
+                TizenLog.Error("Could not get an Evas object.");
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Sets an engine associated with this view.
+        /// </summary>
+        public void SetEngine(FlutterEngine engine)
+        {
+            Engine = engine;
+        }
+
+        /// <summary>
+        /// Terminates the running view and the associated engine.
+        /// </summary>
+        public void Destroy()
```

Comment:
How about using IDisposable pattern?

Reply:
Using the IDisposable pattern in this context didn't make very much sense to me. My concerns were:

- `ElmFlutterView` will (and should) never be used along with `using`.
- The user will not feel natural about the API `view.Dispose()` because the user has no knowledge of whether the view is associated with any unmanaged resources. On the contrary, the name `view.Destroy()` is consistent with `FlutterDesktopViewDestroy`.
- `ElmFlutterView` is not a direct wrapper of the native Flutter view.

Reply:
What if the `view` object disappears by GC without Destroy () call?

Reply:
That's an undefined behavior but the view will be left running (even if the parent EvasObject disappears) and no C# code will be able to access the view. I think this is better than the view being abruptly (unintentionally) destroyed by the GC but how do you think?

Reply:
I think so. we can register the view internally somewhere, but.. That doesn't make any sense, either.